Extension/src/LanguageServer/configurations.ts

```diff
@@ -2030,6 +2007,43 @@ export class CppProperties {
             const pattern: RegExp = new RegExp(`"[^"]*?(?<="|;)${escapedPath}(?="|;).*?"`, "g");
             const configMatches: string[] | null = curText.match(pattern);
 
+            const expandedPaths: string[] = this.resolveAndSplit([curPath], undefined, this.ExtendedEnvironment, true, false);
+            const incorrectExpandedPaths: string[] = [];
+
+            if (expandedPaths.length > 0) {
+                if (this.rootUri) {
+                    for (const [index, expandedPath] of expandedPaths.entries()) {
+
+                        if (expandedPath.includes("${workspaceFolder}")) {
+                            expandedPaths[index] = this.resolvePath(expandedPath, false);
+                        } else {
+                            expandedPaths[index] = this.resolvePath(expandedPath);
+                        }
+
+                        const checkPathExists: any = util.checkPathExistsSync(expandedPaths[index], this.rootUri.fsPath + path.sep, isWindows, false);
+                        if (!checkPathExists.pathExists) {
+                            // If there are multiple paths, store any non-existing paths to squiggle later on.
+                            incorrectExpandedPaths.push(expandedPaths[index]);
+                        }
+                    }
+                }
+            } else {
+                continue;
+            }
+
+            const pathExists: boolean = incorrectExpandedPaths.length === 0;
+
+            for (const [index, expandedPath] of expandedPaths.entries()) {
+                // Normalize path separators.
+                if (path.sep === "/" && expandedPaths.length > 0) {
+                    expandedPaths[index] = expandedPath.replace(/\\/g, path.sep);
+                } else {
+                    expandedPaths[index] = expandedPath.replace(/\//g, path.sep);
+                }
+            }
+
+            // Iterate through the text and apply squiggles.
```

Comment:
Is this comment at the wrong spot? Usually "iterate" comments would occur right before a loop.

Reply:
It looks like that comment is pre-existing.

Reply:
![image](https://github.com/microsoft/vscode-cpptools/assets/111317156/dc62120b-23ca-4676-823b-7f39d0f5baf0)

This was an existing comment before my changes. It seems appropriate.

Reply:
ok